middle_end/flambda2/to_cmm/to_cmm_env.ml

```diff
@@ -360,6 +362,16 @@ let classify effs =
      effectful expressions such as function calls. *)
   | No_effects, No_coeffects -> Pure
 
+let is_inlinable_box effs ~extra =
+  (* [effs] is the effects and coeffects of some primitive operation, arising
+     either from the primitive itself or its arguments. If this is a boxing
+     operation (as indicated by [extra]), then we want to inline the box, but
+     this involves moving the arguments, so they must be pure (or at most
+     generative). *)
+  match (effs : Effects_and_coeffects.t), (extra : extra_info option) with
+  | (Only_generative_effects _, No_coeffects), Some Box -> true
```

Comment:
I believe that we won't get `Pure` boxed allocations, as the boxing itself will always contain generative effects, but it would read better (and match the comment above) if we allowed `Pure` effects too.

Reply:
Fixed